tokendito/helpers.py

```diff
@@ -312,6 +312,25 @@ def validate_saml_response(html):
     return xml
 
 
+def validate_okta_aws_app_url(url):
+    """Validate whether a given URL is a valid AWS app URL in Okta.
+
+    :param url: string
+    :return: bool. True if valid, False otherwise
+    """
+    logging.debug('Will try to match \'{}\' to a valid URL'.format(url))
+
+    url = urlparse(url)
+    # Here, we could also check url.netloc against r'.*\.okta(preview)?\.com$'
+    # but Okta allows the usage of custome URLs such as login.acme.com
+    if url.scheme == 'https' and \
+       re.match(r'^/home/amazon_aws/\w{20}/\d{3}$', url.path) is not None:
```

Comment:
Do custom URLs always follow the /home/amazon_aws/<app_id> format or does it make sense to change our logging.error on line 447 to this instead: "Please provide the Okta Application URL in the standard format and try again."

Reply:
custom URLs only apply to the top level domain, e.g. https://login.company.tld/. The URI is standardized to `/home/amazon_aws/appid`